src/Domain/Services/ExtendRestApi.php

```diff
@@ -0,0 +1,129 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\Domain\Services;
+
+use Throwable;
+
+/**
+ * Service class to provide utility functions to extend REST API.
+ */
+class ExtendRestApi {
+	/**
+	 * Valid endpoints to extend
+	 *
+	 * @var array
+	 */
+	private $endpoints = [ 'item' ];
+
+	/**
+	 * An endpoint that validates registration method call
+	 *
+	 * @param string   $endpoint The endpoint to extend.
+	 * @param string   $namespace Plugin namespace.
+	 * @param callable $schema_callback Callback executed to add schema data.
+	 * @param callable $data_callback Callback executed to add endpoint data.
+	 */
+	private function validate_endpoint_data( $endpoint, $namespace, $schema_callback, $data_callback ) {
+		if ( ! is_string( $namespace ) ) {
+			$this->throw_exception(
+				'You must provide a plugin namespace when extending a Store REST endpoint.'
+			);
+		}
+
+		if ( ! is_string( $endpoint ) || ! in_array( $endpoint, $this->endpoints, true ) ) {
+			$this->throw_exception(
+				'You must provide a valid Store REST endpoint to extend, valid endpoints are: ' . implode( ', ', $this->endpoints )
+			);
+		}
+
+		if ( ! is_callable( $schema_callback ) ) {
+			$this->throw_exception(
+				'$schema_callback must be a callable function.'
+			);
+		}
+
+		if ( ! is_callable( $data_callback ) ) {
+			$this->throw_exception(
+				'$data_callback must be a callable function.'
+			);
+		}
+	}
```

Comment:
This function is going to be used internally by other extend methods, it does the namespace verification, the endpoint verification (not needed if we're going to create wrappers for each endpoint), and callback verification.